mlr-org/gallery/technical/2023-11-09-memory-disk/index.qmd

```diff
@@ -0,0 +1,87 @@
+---
+title: "Why is mlr3 Eating my Disk / RAM?"
+description: |
+    Possible explanations for exuberant memory usage of mlr3 objects.
+categories:
+  - tuning
+  - classification
+author:
+  - name: Marc Becker
+    url: https://github.com/be-marc
+  - name: Sebastian Fischer
+    url: https://github.com/sebffischer
+date: 2023-11-09
+knitr:
+  opts_chunk:
+    R.options:
+      datatable.print.nrows: 6
+      datatable.print.trunc.cols: TRUE
+---
+
+{{< include ../../_setup.qmd >}}
+
+```{r}
+#| include: false
+lgr::get_logger("mlr3")$set_threshold("warn")
+```
+
+
+When serializing `mlr3` objects, it can happen that their size is much larger than expected.
+While we are trying our best to keep such cases from happening, there are things that are beyond our control.
+This gallery post serves as a technical trouble-shooting guide that covers various issues and offers solutions where possible.
+We will update this post as new problems come to our attention.
+Note that while some of these issues might seem neglibile, they can cause serious problems when running large benchmrk experiments, e.g. using `mlr3batchmark`.
+
+## Avoid Installating Packages With Source References
+
+Some objects in `mlr3` have parameters that can be functions.
+One example for that is `po("colapply")`'s `applicator` parameter.
+
+```{r, output = FALSE}
+library("mlr3verse")
+library("pryr")
+po_center = po("colapply", applicator = function(x) x - mean(x))
+```
+
+Because `Learner`s store the hyperparameters that were used for training in their `$state`, it is important to ensure that their size is small.
+One cause for large sizes of parameter values is the presence of source references in the function's attributes.
+Source references are kept when installing packages with the `--with-keep.source` option.
+Note that this option is enabled by default when installing packages with `renv`.
```

Comment:
Mention that installing with source refs is not the normal case but  I would point out renv earlier. How can I check that mlr3 was installed with source refs?